modules/HapticGlove_module/src/RobotInterface.cpp

```diff
@@ -0,0 +1,1216 @@
+/**
+ * @file RobotInterface.cpp
+ * @authors Kourosh Darvish <kourosh.darvish@iit.it>
+ * @copyright 2020 iCub Facility - Istituto Italiano di Tecnologia
+ *            Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ * @date 2020
+ */
+
+#include <algorithm>
+#include <iterator>
+#include <limits>
+
+// iDynTree
+#include <iDynTree/Core/Utils.h>
+
+// teleoperation
+#include <ControlHelper.hpp>
+#include <RobotInterface.hpp>
+#include <Utils.hpp>
+
+using namespace HapticGlove;
+
+bool RobotInterface::configure(const yarp::os::Searchable& config,
+                               const std::string& name,
+                               const bool& rightHand,
+                               const bool& isMandatory)
+{
+    m_rightHand = rightHand;
+    m_logPrefix = "RobotInterface::";
+    m_logPrefix += m_rightHand ? "RightHand:: " : "LeftHand:: ";
+
+    m_isMandatory = isMandatory;
+
+    // robot name: used to connect to the robot
+    std::string robot;
+    robot = config.check("robot", yarp::os::Value("icubSim")).asString();
+
+    if (!YarpHelper::getVectorFromSearchable(config, "axis_list", m_actuatedAxisNames))
+    {
+        yError() << m_logPrefix << "unable to get  axis_list from the config file.";
+        return false;
+    }
+    m_noActuatedAxis = m_actuatedAxisNames.size();
+
+    if (!YarpHelper::getVectorFromSearchable(config, "all_axis_list", m_allAxisNames))
+    {
+        yError() << m_logPrefix << "unable to get  all_axis_list from the config file..";
+        return false;
+    }
+
+    // // 5*3 analog sensors + 1 encoder thumb oppose + 1 encoder hand_finger
+    if (!YarpHelper::getVectorFromSearchable(config, "joint_list", m_allJointNames))
+    {
+        yError() << m_logPrefix << "unable to get joint_list from the config file.";
+        return false;
+    }
+
+    m_noAllAxis = m_allAxisNames.size();
+
+    m_noAllJoints = m_allJointNames.size();
+
+    // add the list of axis and associated analog sensors
+    std::vector<std::string> analogList;
+    if (!YarpHelper::getVectorFromSearchable(config, "analog_list", analogList))
+    {
+        yError() << m_logPrefix << "unable to get analog_list from config file.";
+        return false;
+    }
+
+    m_noAnalogSensor = analogList.size();
+
+    std::vector<bool> jointFeedbackIsAnalog(m_noAllJoints, false);
+    for (size_t i = 0; i < m_allJointNames.size(); i++)
+    {
+        if (find(std::begin(analogList), std::end(analogList), m_allJointNames[i])
+            != std::end(analogList))
```

Comment:
Suggestion. The ``find`` moves linearly across the ``analogList`` vector (hence the complexity is linear) once for every joint. You can save some time if the ``find`` complexity was constant. This is possible using some more complex data structure like [``std::unordered_set``](https://en.cppreference.com/w/cpp/container/unordered_set/find). Basically, the data is saved using some hashing mechanism that allows finding an entry in constant time. This may also come useful if you have to scan the vector multiple times. It is really not a big deal here since you are in the configuration phase, but it might be helpful in other cases.

Reply:
I see your point. Thank you for the explanation. As you mentioned here, we are in the configuration time and it should be OK. Moreover, the order of elements of the elements of the `analogList` is important in our case, and as the elements are not sorted in any particular order in [`unordered_set`](https://en.cppreference.com/w/cpp/container/unordered_set) it may cause problem. As you can see in https://github.com/robotology/walking-teleoperation/blob/fc89789c4fabb663a023cae91fd3c22e512bb5b0/modules/HapticGlove_module/src/RobotInterface.cpp#L125 , we are considering the index of the `analogList` for future data association.